ESP8266/ESP8266.cpp

```diff
@@ -82,10 +82,31 @@ bool ESP8266::dhcp(bool enabled, int mode)
         && _parser.recv("OK");
 }
 
-bool ESP8266::connect(const char *ap, const char *passPhrase)
+int ESP8266::connect(const char *ap, const char *passPhrase)
```

Comment:
If you're using NSAPI_ERROR codes, you can make this explicitly return nsapi_error_t, unless there's some reason to avoid including the header from your header.